craft_application/commands/lifecycle.py

```diff
@@ -321,7 +323,21 @@ def run(self, parsed_args: argparse.Namespace) -> None:
         """Run the clean command."""
         super().run(parsed_args)
 
-        self._services.lifecycle.clean(parsed_args.parts)
+        if self._should_clean_instances(parsed_args):
+            self._services.provider.clean_instances()
+        else:
+            self._services.lifecycle.clean(parsed_args.parts)
+
+    @override
+    def run_managed(self, parsed_args: argparse.Namespace) -> bool:
+        # "clean" should run managed if cleaning specific parts.
+        # otherwise, should run on the host to clean the build provider.
+        return not self._should_clean_instances(parsed_args)
+
+    @staticmethod
+    def _should_clean_instances(parsed_args: argparse.Namespace) -> bool:
+        # Note: in the future this will also take into account destructive mode.
```

Comment:
Would this be any more complex than returning False if we're in destructive mode?

Reply:
Nope! At least I hope not hehe